Makefile

```diff
@@ -105,12 +105,17 @@ W32ODIRTILES = objwin/tiles
 DDIR = .deps
 
 OS  = $(shell uname -s)
+
+# Expand at reference time to avoid recursive reference
+OS_COMPILER := $(CXX)
+OS_LINKER := $(LD)
+
 ifdef CCACHE
-  CXX = ccache $(CROSS)g++
-  LD  = ccache $(CROSS)g++
+  CXX = ccache $(OS_COMPILER)
+  LD  = ccache $(OS_LINKER)
```

Comment:
So, uh, is there a reason you're blowing away the `$(CROSS)` reference for the ccache build, but not for the non-ccache build below? I'm not sure which is correct, but you should either assume the cross prefix is part of the original `$(CXX)`, or assume it isn't.